File: frontend/src/types/channel.ts
import type { Video } from "./video";

export interface Channel {
    id: string;
    name: string;
    description?: string;
    bannerUrl?: string;
    avatarUrl?: string;
    ownerAvatarUrl?: string;
    subscribersCount: number;
    notificationEnabled: boolean;
    ownerId: string;
    ownerUsername: string;
    createdAt: string;
    featuredVideoId?: string | null;
    featuredVideo?: Video | null;
}

export interface CreateChannelRequest {
    name: string;
    handle: string;
    description?: string;
}

export interface Subscription {
    id: string;
    channelId: string;
    channelName?: string;
    name?: string;
    notificationEnabled?: boolean;
    channelAvatarUrl?: string | null;
    avatarUrl?: string | null;
    ownerAvatarUrl?: string | null;
    hasNewVideos?: boolean;
}

File: backend/VidFlow/Models/Channel.cs
namespace VidFlow.Models
{
    public class Channel
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string? BannerUrl { get; set; }
        public int SubscribersCount { get; set; } = 0;

        public Guid OwnerId { get; set; }

        public User Owner { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Subscription> Subscriptions { get; set; } = new List<Subscription>();

        public ICollection<Video> Videos { get; set; } = new List<Video>();

        public Guid? FeaturedVideoId { get; set; }
        public Video? FeaturedVideo { get; set; }
    }
}
